waku/factory/node_factory.nim

```diff
@@ -113,6 +113,12 @@ proc initNode(
 
 ## Mount protocols
 
+proc getNumShardsInNetwork*(conf: WakuNodeConf): uint32 =
+  if conf.numShardsInNetwork != 0:
+    return conf.numShardsInNetwork
+  # If conf.numShardsInNetwork is not set, use the number of shards configured as numShardsInNetwork
+  return uint32(max(conf.shards) + 1)
```

Comment:
is it safe to assume that number of shards in the network is the configured shards count (in case static sharding is being used)? 
what if i am running a node which only supports few of the shards that are there in the network in case of static sharding. will this assumption break any functionality?

Reply:
We initially had that the number of shards configured was the shard count, but we got to edge cases in which if someone subscribed only to shards 4 and 5 for example, we would get that it subscribed to shards 4 and 5 in a network with 2 shards and made no sense.

So I modified it to use the highest number of a shard configured + 1 as number of shards in the network. So if someone subscribes only to shards 4 and 5 and doesn't set manually the number of shards in the network, we deduce that there's 6 shards in the network (0 to 5)

And with this other approach, functionality isn't broken :)

Reply:
> So I modified it to use the highest number of a shard configured + 1 as number of shards in the network. So if someone subscribes only to shards 4 and 5 and doesn't set manually the number of shards in the network, we deduce that there's 6 shards in the network (0 to 5)

But that seems like a wrong assumption that could lead to issues. Think of status node , it subscribes only to shards 1,32,64,128,256 as of now. But number of shards in the network is all available 1024 as per the static sharding indicated in the [spec](https://github.com/waku-org/specs/blob/master/standards/core/relay-sharding.md#static-sharding).
This assumption may not have an impact as of now, but was wondering if it will cause future issues. Instead if number of shards is not configured, why not consider it as 1024 since all 1024 can be used.

Reply:
> Instead if number of shards is not configured, why not consider it as 1024 since all 1024 can be used.

Makes sense! Yes, I actually think it's a better solution and will lead to more nodes "agreeing" on the number of shards in a network, as it doesn't depend on their shard subscriptions.

Ideally, nodes should configure manually the number of shards in the network they use. But if static sharding specification limits it to 1024, then it makes sense to use 1024 as a default.

Will update the PR :) Thanks so much!

Reply:
> Ideally, nodes should configure manually the number of shards in the network they use. But if static sharding specification limits it to 1024, then it makes sense to use 1024 as a default.

Yep, but that is only in case of auto-sharding. In case of static sharding, nodes don't specify number of shards which is why 1024 made sense.